R/densityClust.R

```diff
@@ -126,14 +126,22 @@ distanceToPeak <- function(distance, rho) {
 #' 
 #' @export
 #' 
-estimateDc <- function(distance, neighborRateLow=0.01, neighborRateHigh=0.02) {
+estimateDc <- function(distance, neighborRateLow=0.01, neighborRateHigh=0.02, seed = NULL) {
    # This implementation uses binary search instead of linear search.
    
    size <- attr(distance, 'Size')
    # If size is greater than 448, there will be >100000 elements in the distance
    # object. Subsampling to 100000 elements will speed performance for very 
    # large dist objects while retaining good accuracy in estimating the cutoff
    if(size > 448) {
+      # ensure the consistency of clustering result
+      if(! is.null(seed)) {
+        set.seed(2017) 
+      }
+      else {
+        warning('random sampling 100128 points, please pass a number for the seed argument to ensure consistency')
+      }
+
```

Comment:
fixed

Reply:
Seed still shouldn't be an argument to the function in any way and the function should not set the seed. This should only be done directly by the user by writing set.seed() in the console. The reason for this is that if a function sets a seed it will affect all code using random numbersbeing called after the function...